VTCareService/src/main/java/edu/vt/cs/vtcare/vtcareservice/dao/ProviderDao.java

```diff
@@ -28,14 +42,55 @@ public long persistProvider(Provider provider) throws SQLException {
             statement.setString(8, provider.getSpecialization());
             statement.setInt(9, provider.getYearsOfExperience());
             statement.executeUpdate();
-            long providerId = -1;
+
             ResultSet rs = statement.getGeneratedKeys();
             if (rs.next()) {
                 providerId = rs.getLong(1);
             }
-            return providerId;
         } catch (SQLException e) {
-            throw new SQLException("Encountered problem creating a new provider ", e);
+            System.out.println("Encountered problem creating a new provider " + e);
```

Comment:
Just an observation, why aren't we throwing the exception back to the controller, and printing it out to the console?

Reply:
Not needed as we are handling it. Printing to console is a make up for proper logging.